matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/crypto/tasks/RedactEventTask.kt

```diff
@@ -26,22 +28,31 @@ internal interface RedactEventTask : Task<RedactEventTask.Params, String> {
             val txID: String,
             val roomId: String,
             val eventId: String,
-            val reason: String?
+            val reason: String?,
+            val withRelations: List<String>?
     )
 }
 
 internal class DefaultRedactEventTask @Inject constructor(
         private val roomAPI: RoomAPI,
-        private val globalErrorReceiver: GlobalErrorReceiver
+        private val globalErrorReceiver: GlobalErrorReceiver,
+        private val homeServerCapabilitiesService: HomeServerCapabilitiesService,
 ) : RedactEventTask {
 
     override suspend fun execute(params: RedactEventTask.Params): String {
+        val withRelations = if (homeServerCapabilitiesService.getHomeServerCapabilities().canRedactEventWithRelations &&
+                !params.withRelations.isNullOrEmpty()) {
+            params.withRelations
+        } else {
+            null
+        }
+
         val response = executeRequest(globalErrorReceiver) {
             roomAPI.redactEvent(
                     txId = params.txID,
                     roomId = params.roomId,
                     eventId = params.eventId,
-                    reason = if (params.reason == null) emptyMap() else mapOf("reason" to params.reason)
+                    body = EventRedactBody(params.reason, withRelations, withRelations)
```

Comment:
Can we use named argument here for `EventRedactBody` to clearly see what parameter it refers to?

Reply:
https://github.com/vector-im/element-android/pull/7988/commits/52d0b826f494e2c19e2b2b8818f853ac0b03596f